src/tokenizer.ts

```diff
@@ -1494,37 +1494,78 @@ export class Tokenizer extends DiagnosticEmitter {
   }
 
   readDecimalFloat(): f64 {
-    // TODO: numeric separators (parseFloat can't handle these)
     var text = this.source.text;
-    var pos = this.pos;
     var end = this.end;
-    var start = pos;
-    while (pos < end && isDecimalDigit(text.charCodeAt(pos))) {
-      ++pos;
-    }
-    if (pos < end && text.charCodeAt(pos) == CharCode.DOT) {
-      ++pos;
-      while (pos < end && isDecimalDigit(text.charCodeAt(pos))) {
-        ++pos;
-      }
+    var start = this.pos;
+    var hasSep = false;
+
+    hasSep = this.readDecimalFloatPartial(false) || hasSep;
```

Comment:
Having looked again, I think that it makes sense to use an integer counter instead of a boolean value:
```ts
var sepCount = 0;

sepCount += this.readDecimalFloatPartial(false);
...

if (sepCount) result = result.replaceAll("_", "");
return parseFloat(result);


```